src/gt4py/next/iterator/transforms/constant_folding.py

```diff
@@ -0,0 +1,49 @@
+# GT4Py - GridTools Framework
+#
+# Copyright (c) 2014-2023, ETH Zurich
+# All rights reserved.
+#
+# This file is part of the GT4Py project and the GridTools framework.
+# GT4Py is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or any later
+# version. See the LICENSE.txt file at the top-level directory of this
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from gt4py.eve import NodeTranslator
+from gt4py.next.iterator import embedded, ir, ir_makers as im
+
+
+class ConstantFolding(NodeTranslator):
+    @classmethod
+    def apply(cls, node: ir.Node):
+        return cls().visit(node)
+
+    def visit_FunCall(self, node: ir.FunCall):
+        new_node = self.generic_visit(node)
+
+        if (
+            isinstance(new_node.fun, ir.SymRef)
+            and new_node.fun.id == "if_"
+            and isinstance(new_node.args[0], ir.Literal)
+        ):  # if_(True, true_branch, false_branch) -> true_branch
+            if new_node.args[0].value == "True":
+                new_node = new_node.args[1]
+            else:
+                new_node = new_node.args[2]
+
+        if (
+            isinstance(new_node, ir.FunCall)
+            and isinstance(new_node.fun, ir.SymRef)
+            and len(new_node.args) > 0  # for zero dims fields
```

Comment:
```suggestion
            and len(new_node.args) > 0
```
In any case this is not needed anymore when you use the whitelist approach.